tests/integration/kubernetes/k8s-debug-console.bats

```diff
@@ -0,0 +1,50 @@
+#!/usr/bin/env bats
+# Copyright (c) 2025 Advanced Micro Devices, Inc.
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+load "${BATS_TEST_DIRNAME}/../../common.bash"
+load "${BATS_TEST_DIRNAME}/tests_common.sh"
+load "${BATS_TEST_DIRNAME}/lib.sh"
+
+# For kata-runtime
+export KATA_HOME="/opt/kata"
+export KATA_RUNTIME="${KATA_HOME}/bin/kata-runtime"
+
+timeout=${timeout:-30}
+
+setup() {
+  pod_name="busybox-base-pod"
+  get_pod_config_dir
+  yaml_file="${pod_config_dir}/pod-busybox-base.yaml"
+  
+  #add_allow_all_policy_to_yaml "${yaml_file}"
+  #expected_name="${pod_name}"
+}
+
+@test "Access and verify that the debug console is working" {
+  # Create pod
+  kubectl apply -f "${yaml_file}"
+
+  # Check pod creation
+  kubectl wait --for=condition=Ready --timeout=${timeout} pod "${pod_name}"
+
+  # Get sandbox ID
+  sandbox_id=$(get_kata_sandbox_id_by_pod_name "${pod_name}")
+  echo "Sandbox ID for pod [${pod_name}]: ${sandbox_id}"
+
+  # Test debug console
+  local kata_agent_path=$(sudo "${KATA_RUNTIME}" exec "${sandbox_id}" which kata-agent)
+  if [[ ! "${kata_agent_path}" =~ "kata-agent" ]]; then
+    echo "ERROR: The debug console could not locate the kata-agent: ${kata_agent_path}" >&2
+    return 1
+  fi
+}
```

Comment:
Testing this is slightly more complicated than you're considering, @ryansavino.

Right now the majority of the tests are running on AKS clusters, which are not the same machine where we're running the tests from, thus we don't actually have access to the install `kata-runtime` nor the VMM socket to just exec into the guest.

In order to get this to work, you'd need to limit such tests on baremetal / self-hosted runners ... but then the majority of those are TEEs, and I don't fully remember whether we allow `exec` into the guest or not.

Reply:
Right on. I was a bit worried that this test wouldn't be applicable in certain workflows. Should I just restrict this to a CoCo TEE specific test?

Reply:
For SNP I know the debug console is enabled in the config for the ci by default. I've manually tested this, and it's working. I think it's the same way for all TEEs.

Reply:
Sure thing. I've now changed this to restrict it to when `K8S_TEST_HOST_TYPE="baremetal"`. This also excludes the non tee jobs as well.